tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -0,0 +1,254 @@
+use super::{span, Format, FormatEvent, FormatFields, FormatTime};
+use crate::field::MakeVisitor;
+use serde::ser::{SerializeMap, Serializer as _};
+use serde_json::Serializer;
+use std::{
+    collections::BTreeMap,
+    fmt::{self, Write},
+};
+use tracing_core::{
+    field::{self, Field},
+    Event,
+};
+use tracing_serde::{AsSerde, WriteAdaptor};
+
+#[cfg(feature = "tracing-log")]
+use tracing_log::NormalizeEvent;
+
+/// Marker for `Format` that indicates that the verbose json log format should be used.
+///
+/// The full format includes fields from all entered spans.
+#[derive(Default, Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Json;
+
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        use tracing_serde::fields::AsMap;
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(WriteAdaptor::new(writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_entry("timestamp", &timestamp)?;
+            serializer.serialize_entry("level", &meta.level().as_serde())?;
+
+            ctx.with_current(|(_, span)| serializer.serialize_entry("span", &span))
```

Comment:
This serializes only the current span in which the event occurred, rather than the span and all its parents (the way the `Format<Full, ...>` formatter does. We might want to have separate "full" and "compact" JSON formatters as well (though I'm fine with adding that in a follow-up).